modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AttachmentAwareClient.java

```diff
@@ -62,14 +67,24 @@ public T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, String kin
 
     private T uploadAndAttachAttachment(T itemToModify, Path fileToAttach, HttpEntity<MultiValueMap<String, Object>> requestEntity) {
         final String self = itemToModify.get_Links().getSelf().getHref();
-        ResponseEntity<T> response = restTemplate.postForEntity(self + ATTACHMENTS_ENDPOINT, requestEntity, getHandledClassType());
+        ResponseEntity<T> response;
```

Comment:
Idea was to have something like this, to prevent this mixed imperative and functional stuff:
```suggestion
        return Optional.ofNullable({
            try {
                restTemplate.postForEntity(self + ATTACHMENTS_ENDPOINT, requestEntity, getHandledClassType());
            } catch (...) {
                return null;
            })
            .filter(res -> res.getStatusCode().is2xxSuccessful())
            .map(ResponseEntity::getBody)
            .orElseGet(...);
```

Reply:
With this solution Logging wouldn't be able if the response is not successful.